### DISCUSSION TITLE
Quantity from value and unit-abbreviation

### DISCUSSION THREAD

**Discussion by igiona** [angularsen/UnitsNet#1252] [Q&A] [answered]
Up to now, I was not able to find a way to create a Quantity based on a value and its unit abbreviation.
For example, (3, "mm") should convert in a Quantity object of type Length and unit LengthUnit.Millimeter .
Looking at the code, it seems the the Unit enum is, in a way or another, always required.

Am I missing something?

I'm aware that there might be ambigues abbreviations, but for the majority of the cases this should be working fine though.
Eventually one could restrict such converted to SI units?
It would be extremally useful to reduce the amount of metadata one has to store, in order to create a quantity value at runtime.

**Comment by angularsen** [comment, marked as answer]
Hi, sorry for the late reply.

If you are looking to serialize data to storage and read it back, I would recommend this instead:
```cs
IQuantity quantity = Quantity.From(value: 3, quantityName: "Length", unitName: "Centimeter");

if (Quantity.TryFrom(value: 3, quantityName: "Length", unitName: "Centimeter", out IQuantity? quantity2))
{
}
```

If you still need to store and read unit abbreviations, which I recommend against, then I just merged a naive feature for this you could try.
#1265 

There is a relatively high chance of failure due to multiple units (across all quantities) having the same abbreviation.

New example in the README:
```c#
// Length with unit LengthUnit.Centimeter
IQuantity quantity = Quantity.FromUnitAbbreviation(3, "cm");

if (Quantity.TryFromUnitAbbreviation(3, "cm", out IQuantity? quantity2))
{
}
```


Let me know how this works for you.

Nuget should be out shortly.

[Release UnitsNet/5.21.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/UnitsNet%2F5.21.0)

**Reply by igiona** [reply]
Many thanks for looking into this and providing a solution already 👏 !

I think that the current implementation will work just fine for most of my use-cases in which I need to manually serialize/maintain some metadata files and I need to deserialize them programmatically. I'm going to test it soon.

In case of abbreviation collisions and/or when the serialization is done programmatically, I agree with your preferred approach 👍
